skiko/src/jvmMain/kotlin/org/jetbrains/skiko/FrameDispatcher.kt

```diff
@@ -23,14 +22,15 @@ class FrameDispatcher(
         onFrame
     )
 
-    private var needFrame = CompletableDeferred<Unit>()
+    private var needFrame = false
+    private val channel = Channel<Unit>(Channel.CONFLATED)
 
     private val job = scope.launch {
         while (true) {
-            needFrame.await()
-            needFrame = CompletableDeferred()
+            if (!needFrame)
+                channel.receive()
+            needFrame = false
             onFrame()
-            yield()
```

Comment:
We need this `yield`.

If we have animation then `scheduleFrame` will be called inside `onFrame` and we will never free the Swing dispatcher.